ci/bootstrap.py

```diff
@@ -0,0 +1,336 @@
+                        ])
+
+                if j._mount_docker_socket:
+                    mount_options.extend(['-v', '/var/run/docker.sock:/var/run/docker.sock'])
+
+                main_cid, main_ok = await docker_run(
+                    'docker', 'run', '-d',
+                    *env_options, *mount_options, j._image, *j._command)
+                print(f'{j._index}: {job_name}/main: {main_cid} {"OK" if main_ok else "FAILED"}')
+            else:
+                main_ok = False
+                print(f'{j._index}: {job_name}/main: SKIPPED: input failed')
+
+            if j._output_files:
+                if main_ok:
+                    copy_script = 'set -ex\n'
+                    for src, dest in j._output_files:
+                        assert dest.startswith(prefix), (prefix, dest)
+                        dest = f'/shared{dest[len(prefix):]}'
+                        if dest.endswith('/'):
+                            copy_script = copy_script + f'mkdir -p {dest}\n'
+                        else:
+                            copy_script = copy_script + f'mkdir -p {os.path.dirname(dest)}\n'
+                        copy_script = copy_script + f'cp -a {src} {dest}\n'
+                    output_cid, output_ok = await docker_run(
+                        'docker', 'run', '-d', '-v', f'{root}/shared:/shared', '-v', f'{job_root}/io:/io', 'ubuntu:18.04', '/bin/bash', '-c', copy_script)
+                    print(f'{j._index}: {job_name}/output: {output_cid} {"OK" if output_ok else "FAILED"}')
+                else:
+                    output_ok = False
+                    print(f'{j._index}: {job_name}/output: SKIPPED: main failed')
+            else:
+                output_ok = True
+
+            j._succeeded = (input_ok and main_ok and output_ok)
+            j._done = True
+
+
+class Branch(Code):
+    def __init__(self, owner, repo, branch, sha, extra_config):
+        self._owner = owner
+        self._repo = repo
+        self._branch = branch
+        self._sha = sha
+        self._extra_config = extra_config
+
+    def short_str(self):
+        return f'br-{self._owner}-{self._repo}-{self._branch}'
+
+    def repo_dir(self):
+        return '.'
+
+    def branch_url(self):
+        return f'https://github.com/{self._owner}/{self._repo}'
+
+    def config(self):
+        config = {
+            'checkout_script': self.checkout_script(),
+            'branch': self._branch,
+            'repo': f'{self._owner}/{self._repo}',
+            'repo_url': self.branch_url(),
+            'sha': self._sha
+        }
+        config.update(self._extra_config)
+        return config
+
+    def checkout_script(self):
+        return f'''
+{clone_or_fetch_script(self.branch_url())}
+
+git checkout {shq(self._sha)}
+'''
+
+
+async def main():
+    await kube.config.load_kube_config()
+
+    parser = argparse.ArgumentParser(description='Create initial Hail as a service account.')
```

Comment:
This does much more than create a HaaS account. I think this is "Execute build.yaml without an extant CI"